src/__tests__/SearchData.test.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import TableContainer from '../components/pages/Table/TableContainer';
+import { legacy_createStore as createStore } from 'redux';
+import { Provider } from 'react-redux';
+
+import reducers from '../state/reducers';
+
+//anywhere you test antd you must add this window.matchMedia
+window.matchMedia =
+  window.matchMedia ||
+  function () {
+    return {
+      matches: false,
+      addListener: function () {},
+      removeListener: function () {},
+    };
+  };
+
+//creates store for testing redux can be moved to separate component if needed and imported in
+function renderWithRedux(
+  ui,
+  { initialState, store = createStore(reducers, initialState) } = {}
+) {
+  return {
+    ...render(<Provider store={store}>{ui}</Provider>),
+    store,
+  };
+}
+
+beforeEach(() => {
+  renderWithRedux(<TableContainer />);
+});
+
+describe('searching through data shows correct filtered data, advanced search', () => {
```

Comment:
This verbose approach to description proves to be a priceless boon when it comes to figuring things out later -- as developers, we often have a tendency toward brevity, but in this case really detailing what you're doing is absolutely essential to pinpointing aberrations in functionality later on (which is the entire purpose of testing).